subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/verification/ChecksumAndSignatureVerificationOverride.java

```diff
@@ -173,121 +183,30 @@ public void artifactsAccessed(String displayName) {
-                            .map(FailureWrapper::getFailure)
-                            .map(this::extractFailedFilePaths)
-                            .forEach(affectedFiles::add);
-                        hasFatalFailure.set(true);
-                        formatter.node("On artifact " + key + " ");
-                        if (failures.size() == 1) {
-                            FailureWrapper firstFailure = failures.iterator().next();
-                            explainSingleFailure(formatter, maybeCompromised, hasMissing, failedSignatures, hasUntrusted, firstFailure);
-                        } else {
-                            explainMultiFailure(formatter, maybeCompromised, hasMissing, failedSignatures, hasUntrusted, failures);
-                        }
-                    });
-                formatter.endChildren();
-                formatter.blankLine();
-                if (maybeCompromised.get()) {
-                    formatter.node("This can indicate that a dependency has been compromised. Please carefully verify the ");
-                    if (failedSignatures.get()) {
-                        formatter.append("signatures and ");
-                    }
-                    formatter.append("checksums.");
-                }
-                if ((hasMissing.get() || hasUntrusted.get()) && !maybeCompromised.get()) {
-                    // the else is just to avoid telling people to use `--write-verification-metadata` if we suspect compromised dependencies
-                    formatter.node("If the artifacts are trustworthy, you will need to update the gradle/verification-metadata.xml file by following the instructions at " + documentationRegistry.getDocumentationFor("dependency_verification", "sec:troubleshooting-verification"));
-                }
-                if (!affectedFiles.isEmpty()) {
-                    formatter.blankLine();
-                    formatter.node("These files failed verification:");
-                    formatter.startChildren();
-                    for (String affectedFile : affectedFiles) {
-                        formatter.node(affectedFile);
-                    }
-                    formatter.endChildren();
-                    formatter.blankLine();
-                    formatter.node("GRADLE_USER_HOME = " + gradleUserHome);
-                }
-                String message = formatter.toString();
+                // There are fatal failures, but not necessarily on all artifacts so we first filter out
+                // the artifacts which only have not fatal errors
+                failures.asMap().entrySet().removeIf(entry -> {
+                    Collection<RepositoryAwareVerificationFailure> value = entry.getValue();
+                    return value.stream().noneMatch(wrapper -> wrapper.getFailure().isFatal());
+                });
+                DependencyVerificationReportWriter.Report report = reportWriter.generateReport(displayName, failures);
```

Comment:
There's a technical limitation in verification now, which makes it "not quite correct": errors can be collected for different configurations concurrently, instead of "per configuration", and we don't have a way to put them in the correct bucket. I would be rather keen on fixing this at some point, than fixing the synchronization itself, which doesn't really hurt because if you hit this you know you have a problem :)